src/aliceVision/matchingImageCollection/ImageCollectionMatcher_generic.cpp

```diff
@@ -85,6 +86,37 @@ void ImageCollectionMatcher_generic::Match(
 
       IndMatches vec_putatives_matches;
       matcher.Match(_f_dist_ratio, regionsJ, vec_putatives_matches);
+
+      if (_useCrossMatching)
+      {
+        // Initialize the matching interface
+        matching::RegionsDatabaseMatcher matcherCross(randomNumberGenerator, _matcherType, regionsJ);  
+
+        IndMatches vec_putatives_matches_cross;
+        matcherCross.Match(_f_dist_ratio, regionsI, vec_putatives_matches_cross);
+
+        //Create a dictionnary of matches indexed by their pair of indexes
+        std::map<std::pair<int, int>, IndMatch> check_matches;
```

Comment:
You could use existing code for this. See here https://github.com/alicevision/AliceVision/blob/develop/src/aliceVision/matching/filters.hpp.  Symmetric matches filtering

Reply:
- This function requires multiple conversions before input, and after output.
- It requires to build 3 vectors the size of the features sets even if the amount of matches is very low

Reply:
I have not looked into the details but the ability to use nndistanceratio or symmetric or combine both may be interesting to expose.